galsim/utilities.py

```diff
@@ -296,6 +296,25 @@ def _convertPositions(pos, units, func):
 
     return pos
 
+def _lin_approx_err(x, f, i):
+    """Error as \int abs(f(x) - approx(x)) when using ith data point to make piecewise linear
+    approximation."""
+    xleft, xright = x[:i+1], x[i:]
+    fleft, fright = f[:i+1], f[i:]
+    xi, fi = x[i], f[i]
+    mleft = (fi-f[0])/(xi-x[0])
+    mright = (f[-1]-fi)/(x[-1]-xi)
+    f2left = f[0]+mleft*(xleft-x[0])
+    f2right = fi+mright*(xright-xi)
+    return np.trapz(np.abs(fleft-f2left), xleft), np.trapz(np.abs(fright-f2right), xright)
+
+def _lin_approx_split(x, f):
+    """Optimally split a tabulated function into two-part piecewise linear approximation.
+    """
+    errs = [_lin_approx_err(x, f, i) for i in xrange(1, len(x)-1)]
```

Comment:
>I think enabling both algorithms would be reasonable.

Done.  I went with `fast_search=True` as the default option, though I used `fast_search=False` in examples/data/getLSSTBandpasses.py and getSEDs.py, since these don't actually get executed that often.